File: Python/Programmars/Level_3/멀리 뛰기.py
def solution(n):
    dy = [0]*(n+1)
    if n in (1, 2):
        return n
    dy[1] = 1
    dy[2] = 2
    for i in range(3, n+1):
        dy[i] = (dy[i-1] + dy[i-2]) % 1234567
    return dy[n]


print(solution(4))  # 5
print(solution(3))  # 3

File: JavaScript/Programmars/Level_2/멀리 뛰기.js
// 문제: https://programmers.co.kr/learn/courses/30/lessons/12914
const solution = (n) => {
  let answer = 0;
  const dynamic = Array.from({ length: n + 2 }, () => 0);
  dynamic[1] = 1;
  dynamic[2] = 1;
  for (let i = 3; i <= n + 1; i++) {
    dynamic[i] = (dynamic[i - 1] + dynamic[i - 2]) % 1234567;
  }

  return dynamic[n + 1];
};

console.log(solution(4)); // 5
console.log(solution(3)); // 3
